compiler/compile.go

```diff
@@ -400,6 +446,17 @@ func containsColoredFuncLit(decl *ast.FuncDecl, colorsByFunc map[ast.Node]*types
 	return
 }
 
+func containsFuncLit(decl *ast.FuncDecl) (yes bool) {
+	ast.Inspect(decl, func(n ast.Node) bool {
+		if _, ok := n.(*ast.FuncLit); ok {
+			yes = true
+			return false
```

Comment:
I'm not sure I understand this statement, `yes` is the named return, which we set to `true` but then immediately override by returning `false`?

Reply:
`ast.Inspect` is an iterator that expects a function that returns whether you want to continue iteration in that part of the AST. Once we find a function literal, we return false to terminate iteration.

I copied the helper function from above and changed it slightly, since we're looking for any function literal, not just colored function literals.

Reply:
😓 my bad, I misread this, thanks for the precision.